### PR TITLE
Writable Snapshot DataSource Added

### PR DESCRIPTION
**Pull request by Sakshi-dell**
# Description
A few sentences describing the overall goals of the pull request's commits.
Writable Snapshot datasource added

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| |

# ISSUE TYPE

- Feature Pull Request

##### RESOURCE OR DATASOURCE NAME

Writable_Snapshot Datasource

##### OUTPUT
image

##### ADDITIONAL INFORMATION

```paste below

```
# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [x] I have maintained at least 80% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have maintained backward compatibility

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [x] Unit tests
- [x] Acceptance tests

### CODE REVIEW

**Comment by Sakshi-dell** [comment]
run e2e test WritableSnapshotDatasource

File: powerscale/provider/writable_snapshot_datasource.go [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+}
+
+// WritablesnapshotDatasourceSchema describes the data source arguments.
+func WritablesnapshotDatasourceSchema(ctx context.Context) schema.Schema {
+	return schema.Schema{
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Optional:            true,
+				Computed:            true,
+				Description:         "Path of the writable snapshot.",
```
Comment (shenda1) [inline powerscale/provider/writable_snapshot_datasource.go:87]: ID

File: powerscale/provider/writable_snapshot_datasource.go [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+							Description:         "The Unix Epoch time the writable snapshot was created.",
+							MarkdownDescription: "The Unix Epoch time the writable snapshot was created.",
+						},
+						"state": schema.StringAttribute{
+							Computed:            true,
+							Description:         "Writable Snapshot state.",
+							MarkdownDescription: "Writable Snapshot state.",
+							Validators: []validator.String{
+								stringvalidator.OneOf(
+									"active",
```
Comment (shenda1) [inline powerscale/provider/writable_snapshot_datasource.go:119]: Do we need validator for computed attribute?

File: powerscale/provider/writable_snapshot_datasource_test.go [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+		},
+	})
+}
+
+var writableSnapshotDatasourceConfig = `
+data "powerscale_writable_snapshot" "preq" {
+}
+data "powerscale_writable_snapshot" "test" {
+	filter {
+		path = data.powerscale_writable_snapshot.preq.writable[0].dst_path
```
Comment (shenda1) [inline powerscale/provider/writable_snapshot_datasource_test.go:118]: Can we create this pre-req on fly?

File: powerscale/helper/writable_snapshot_helper.go [hunk 4]
```diff
@@ -45,6 +46,31 @@ func DeleteWritableSnapshot(ctx context.Context, client *client.Client, path str
...
+		}
+		if !state.WritableSnapshotFilter.Dir.IsNull() {
+			writablesnapshots = writablesnapshots.Dir(state.WritableSnapshotFilter.Dir.ValueString())
+		}
+		if !state.WritableSnapshotFilter.Resume.IsNull() {
+			writablesnapshots = writablesnapshots.Resume(state.WritableSnapshotFilter.Resume.ValueString())
+		}
+	}
+	resp, _, err := writablesnapshots.Execute()
+	return resp, err
```
Comment (rounak-adhikary) [inline powerscale/helper/writable_snapshot_helper.go:71]: If no filter is nil, we need to iteratively get all paginated data, rt?

File: powerscale/provider/writable_snapshot_datasource.go [hunk 5]
```diff
@@ -0,0 +1,234 @@
...
+			"filter": schema.SingleNestedBlock{
+				Attributes: map[string]schema.Attribute{
+					"path": schema.StringAttribute{
+						Optional:            true,
+						Description:         "Only list writable snapshots matching this path.",
+						MarkdownDescription: "Only list writable snapshots matching this path.",
+						Validators: []validator.String{
+							stringvalidator.RegexMatches(regexp.MustCompile("^/ifs$|^/ifs/"), "must start with /ifs or /ifs/"),
+							stringvalidator.LengthBetween(4, 4096),
+						},
```
Comment (rounak-adhikary) [inline powerscale/provider/writable_snapshot_datasource.go:163]: This field will conflict with the other filters, rt?

**Review by shenda1 — APPROVED** [review]

**Review by rounak-adhikary — APPROVED** [review]

File: powerscale/provider/writable_snapshot_datasource_test.go (hunk 3 above)
Comment (Sakshi-dell) [reply powerscale/provider/writable_snapshot_datasource_test.go:122]: Added writableSnapshotResourceConfig

**Pull request merged**